File: app/src/main/java/com/carlos/grabredenvelope/util/UpdateInfoParser.java
package com.carlos.grabredenvelope.util;

import android.util.Log;
import android.util.Xml;

import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by 小不点 on 2016/2/20.
 */
public class UpdateInfoParser {

    private final static String TAG="Version";

    public static List<UpdateInfo> getUpdateInfo(InputStream inputStream){

        List<UpdateInfo> infos=null;
        UpdateInfo info=null;

        XmlPullParser parser= Xml.newPullParser();
        try {
            parser.setInput(inputStream, "utf-8");
            int type=parser.getEventType();
            Log.i(TAG, "服务器版本信息");
            while(type!= XmlPullParser.END_DOCUMENT){
                switch (type) {
                    case XmlPullParser.START_DOCUMENT:
                        infos=new ArrayList<UpdateInfo>();
                        break;
                    case XmlPullParser.START_TAG:
                        if (parser.getName().equals("information")) {
                            info=new UpdateInfo();
                        }else if ("versionCode".equals(parser.getName())) {
                            info.setVersionCode(Integer.valueOf(parser.nextText()));
                            Log.i(TAG, info.getVersionCode() + "");
                        }else if ("versionName".equals(parser.getName())) {
                            info.setVersionName(parser.nextText());
                            Log.i(TAG, info.getVersionName());
                        }else if ("apkUrl".equals(parser.getName())) {
                            info.setApkUrl(parser.nextText());
                            Log.i(TAG, info.getApkUrl());
                        }else if ("description".equals(parser.getName())) {
                            info.setDescription(parser.nextText());
                            Log.i(TAG, info.getDescription());
                        }
                        break;
                    case XmlPullParser.END_TAG:
                        if (parser.getName().equals("information")) {
                            infos.add(info);
                        }
                        break;
                    default:
                        break;
                }
                type=parser.next();
            }

        } catch (XmlPullParserException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return infos;
    }

}

File: app/src/main/java/com/carlos/grabredenvelope/util/UpdateInfoParser.kt
package com.carlos.grabredenvelope.util

import android.util.Log
import android.util.Xml
import org.xmlpull.v1.XmlPullParser
import org.xmlpull.v1.XmlPullParserException
import java.io.IOException
import java.io.InputStream
import java.util.*

/**
 * Created by 小不点 on 2016/2/20.
 */
object UpdateInfoParser {

    private val TAG = "Version"

    fun getUpdateInfo(inputStream: InputStream): List<UpdateInfo>? {

        var infos: MutableList<UpdateInfo>? = null
        var info: UpdateInfo? = null

        val parser = Xml.newPullParser()
        try {
            parser.setInput(inputStream, "utf-8")
            var type = parser.eventType
            Log.i(TAG, "服务器版本信息")
            while (type != XmlPullParser.END_DOCUMENT) {
                when (type) {
                    XmlPullParser.START_DOCUMENT -> infos = ArrayList()
                    XmlPullParser.START_TAG -> if (parser.name == "information") {
                        info = UpdateInfo()
                    } else if ("versionCode" == parser.name) {
                        info!!.versionCode = Integer.valueOf(parser.nextText())
                        Log.i(TAG, info.versionCode.toString() + "")
                    } else if ("versionName" == parser.name) {
                        info!!.versionName = parser.nextText()
                        Log.i(TAG, info.versionName)
                    } else if ("apkUrl" == parser.name) {
                        info!!.apkUrl = parser.nextText()
                        Log.i(TAG, info.apkUrl)
                    } else if ("description" == parser.name) {
                        info!!.description = parser.nextText()
                        Log.i(TAG, info.description)
                    }
                    XmlPullParser.END_TAG -> if (parser.name == "information") {
                        if (info != null) {
                            infos!!.add(info)
                        }
                    }
                    else -> {
                    }
                }
                type = parser.next()
            }

        } catch (e: XmlPullParserException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return infos
    }

}